native/src/routes/Settings.tsx

```diff
@@ -31,36 +29,25 @@ const SectionHeader = styled.Text`
 `
 
 const Settings = ({ navigation }: SettingsProps): ReactElement => {
-  const { settings, updateSettings } = useAppContext()
-  const { cityCode, languageCode } = useContext(AppContext)
+  const appContext = useCityAppContext()
   const showSnackbar = useSnackbar()
   const { t } = useTranslation('settings')
+  const { settings } = appContext
 
-  const setSetting = async (
-    changeSetting: (settings: SettingsType) => Partial<SettingsType>,
-    changeAction?: (settings: SettingsType) => Promise<boolean>,
-  ) => {
+  const safeOnPress = (update: () => Promise<void> | void) => () => {
     const oldSettings = settings
-    const newSettings = { ...oldSettings, ...changeSetting(settings) }
-    updateSettings(newSettings)
-
-    try {
-      const successful = changeAction ? await changeAction(newSettings) : true
-
-      if (!successful) {
-        updateSettings(oldSettings)
-      }
-    } catch (e) {
+    update()?.catch(e => {
       log('Failed to persist settings.', 'error')
       reportError(e)
-      updateSettings(oldSettings)
-    }
+      appContext.updateSettings(oldSettings)
+      showSnackbar({ text: t('error:unknownError') })
```

Comment:
:wrench: A more specific error message would be possible and nice, but would need translation of course.

Reply:
I am not sure if this is desirable/helpful for users. Could you perhaps give a suggestion on what would be a better error message in this case? Something like `Failed to apply settings`/`Fehler beim Anwenden der Einstellungen`? Would this really be better/more helpful to users?

Reply:
yes i think this is better than "unknown error", but if you disagree i am fine with leaving it as it is